jbpm-event-emitters/jbpm-event-emitters-kafka/src/main/java/org/jbpm/event/emitters/kafka/KafkaEventEmitter.java

```diff
@@ -0,0 +1,142 @@
+ *  <li>org.jbpm.event.emitters.kafka.boopstrap.servers - kafka server ip, default is localhost:9092</li>
+ *  <li>org.jbpm.event.emitters.kafka.topic.<processes|tasks|cases>. Topic name for subscribing to these events. Defaults are "jbpm-<processes|tasks|cases>-events"</li>
+ * </ul> 
+ */
+public class KafkaEventEmitter implements EventEmitter {
+    private static final String SOURCE_FORMATTER = "/process/%s/%s";
+    private Producer<String, byte[]> producer;
+    private ObjectMapper mapper;
+
+    public KafkaEventEmitter() {
+        this(new KafkaProducer<>(getConfigs()));
+    }
+
+    // allow unit test
+    KafkaEventEmitter(Producer<String, byte[]> producer) {
+        this.producer = producer;
+        mapper = new ObjectMapper()
+                .setDateFormat(new SimpleDateFormat(System.getProperty(
+                        "org.jbpm.event.emitters.kafka.date_format", System.getProperty(
+                                "org.kie.server.json.date_format",
+                                "yyyy-MM-dd'T'HH:mm:ss.SSSZ"))))
+                .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+                .configure(MapperFeature.PROPAGATE_TRANSIENT_MARKER, true);
+    }
+
+    public void deliver(Collection<InstanceView<?>> data) {
+        // no-op
+    }
+
+    public void apply(Collection<InstanceView<?>> data) {
+        if (data == null || data.isEmpty()) {
+            return;
+        }
+
+        for (InstanceView<?> view : data) {
+            String processId;
+            long processInstanceId;
+            String type;
+            String topic;
+            if (view instanceof ProcessInstanceView) {
+                ProcessInstanceView processInstanceView = (ProcessInstanceView) view;
+                topic = "processes";
+                type = "process";
+                processInstanceId = processInstanceView.getId();
+                processId = processInstanceView.getProcessId();
+            } else if (view instanceof TaskInstanceView) {
+                TaskInstanceView taskInstanceView = (TaskInstanceView) view;
+                topic = "tasks";
+                type = "task";
+                processInstanceId = taskInstanceView.getProcessInstanceId();
+                processId = taskInstanceView.getProcessId();
+            } else if (view instanceof CaseInstanceView) {
+                CaseInstanceView caseInstanceView = (CaseInstanceView) view;
+                topic = "cases";
+                type = "case";
+                processInstanceId = caseInstanceView.getId();
+                processId = caseInstanceView.getCaseDefinitionId();
+            } else {
+                throw new UnsupportedOperationException("view " + view.getClass());
+            }
+            try {
+                producer.send(new ProducerRecord<>(getTopic(topic), mapper.writeValueAsBytes(
```

Comment:
Not involved in the tx. We don't want phantom messages in the queue.

Reply:
Transactional approach has been changed. 
In deliver method we prepare the data
In apply we commit
In drop we rollback
This requires the usage of a ThreadLocal variable for the kafka transactional producer